# CASE STUDY #02

### QUESTION

### An Institution wishes to find out their student’s ability in maths, reading and writing skills. The Institution wants to do an exploratory study to check the following information.
1. Find out how many males and females participated in the test.
2. What do you think about the students' parental level of education?
3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course
4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course
5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns                     #IMPORTING REQUIRED LIBRARIES

In [2]:
data=pd.read_csv('StudentsPerformance.csv')  #LOADING THE GIVEN DATASET

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


### 1. Find out how many males and females participated in the test.


In [4]:
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

### 2. What do you think about the students' parental level of education?

In [5]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

#####  64.4 percent of the parents have completed high school level of education.

### 3. Who scores the most on average for math, reading and writing based on ● Gender ● Test preparation course


### 4. What do you think about the scoring variation for math, reading and writing based on ● Gender ● Test preparation course

In [7]:
data1=data.groupby(['gender'])['math score'].mean()
data2=data.groupby(['gender'])['reading score'].mean()
data3=data.groupby(['gender'])['writing score'].mean()
print(data1,data2,data3)

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64 gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64 gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64


#### math score average is highest for males(68.821577), reading score average is highest for females(72.590734) and finally writing score average is highest for females(72.467181)

In [8]:
data4=data.groupby(['test preparation course'])['math score'].mean()
data5=data.groupby(['test preparation course'])['reading score'].mean()
data6=data.groupby(['test preparation course'])['writing score'].mean()
print(data4,data5,data6)

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64 test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64 test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64


### 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.

In [9]:
data.sort_values(by="math score",ascending=False).head(250)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]

#### The top 25% of students as per their math score are displayed.